### ISSUE TITLE
win32eventreactor redundantly specifies QS_ALLINPUT | QS_ALLEVENTS to MsgWaitForMultipleObjects

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5284]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5284|
|Type|defect|
|Created|2011-09-29 13:51:51Z|
|Branch|https://github.com/twisted/twisted/tree/win32er-qsallinput-5284|

According to http://msdn.microsoft.com/en-us/library/windows/desktop/ms684242(v=vs.85).aspx QS_ALLINPUT is a superset of QS_ALLEVENTS.  win32er only needs to specify QS_ALLINPUT.

Attachments:

* [my-twisted-patch.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/3e9/3e9503f76ad80fce60a6cd76062ec43d7d64a635/d5bf14bfe4ce8fc7b9256a069dbe396aeaef28e5.patch) (619 bytes) - added by moijes12 on 2011-12-22 13:56:28Z - QS_ALLINPUT | QS_ALLEVENTS has been changed to QS_ALLINPUT only

<details><summary>Searchable metadata</summary>

```
trac-id__5284 5284
type__defect defect
reporter__exarkun exarkun
priority__low low
milestone__ 
branch__branches_win32er_qsallinput_5284 branches/win32er-qsallinput-5284
branch_author__moijes12 moijes12
status__closed closed
resolution__fixed fixed
component__core core
keywords__win32_easy win32 easy
time__1317304311000000 1317304311000000
changetime__1324579244000000 1324579244000000
version__None None
owner__exarkun exarkun
cc__moijes12
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [33401]) Branching to 'win32er-qsallinput-5284'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/win32er-qsallinput-5284)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @exarkun|
|-|-|

Thanks.  The patch looks good, and buildbot says the code still passes all our tests.  I'll apply it!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [33404]) Merge win32er-qsallinput-5284

Author: moijes12
Reviewer: exarkun
Fixes: [#5284](https://github.com/twisted/twisted/issues/5284)

Remove `QS_ALLEVENTS` from the `MsgWaitForMultipleObjects` call in
`win32eventreactor`, since it is a subset of `QS_ALLINPUT` which is
also being specified.
